### PR TITLE
Move mmd files to a volume if fails to sent to dmci and handle it when dmci is up

### PR DESCRIPTION
**Pull request by shamlymajeed**

### CODE REVIEW

File: mmd_agent/agent.py [hunk 1]
```diff
@@ -41,27 +44,66 @@
...
+        return 200, "Saved in unsent_mmd directory"
+
+    except Exception as e:
+        logger.error(str(e))
+        return 507, "Cannot write xml data to cache file"
+
+
 # Send the mmd file to the dmci
+# Retry decorator reruns the method 'send_to_dmci' if an exception occurs.
+@retry(requests.exceptions.RequestException, delay=4, tries=1, backoff=2)
```
Comment (johtoblan) [inline mmd_agent/agent.py]: This delay is in seconds? backoff does not make sense with tries = 1 I think?

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 requests>=2.22
-pyyaml>=5.1
\ No newline at end of file
+pyyaml>=5.1
+retry>=0.9.2
```
Comment (johtoblan) [inline requirements.txt]: I am a bit unsure of this retry library. Last release was in 2016.

**Review by johtoblan — CHANGES_REQUESTED** [review]
Remove retry-dependency

File: mmd_agent/handler.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    dmci_url, unsent_mmd_path = read_config()
+
+    try:
+        while True:
+            if os.path.exists(unsent_mmd_path):
+                for filename in os.listdir(unsent_mmd_path):
+                    mmd_path = os.path.join(unsent_mmd_path, filename)
+                    if os.path.isfile(mmd_path):
+                        main(mmd_path, dmci_url)
+            time.sleep(10)  # Sleep for 10 seconds before checking again
```
Comment (johtoblan) [inline mmd_agent/handler.py]: Should sleep be longer?

Comment (shamlymajeed) [reply mmd_agent/handler.py]: should we make it 10 minutes or more?

Comment (johtoblan) [reply mmd_agent/handler.py]: Lets make it 30 minutes.

**Review by johtoblan — APPROVED** [review]

**Pull request merged**